src/ucp/core/ucp_ep.c

```diff
@@ -448,12 +448,14 @@ ucs_status_t ucp_worker_create_mem_type_endpoints(ucp_worker_h worker)
                                               UCP_EP_INIT_FLAG_INTERNAL,
                                               ep_name,
                                               &worker->mem_type_ep[mem_type]);
-        UCS_ASYNC_UNBLOCK(&worker->async);
         if (status != UCS_OK) {
+            UCS_ASYNC_UNBLOCK(&worker->async);
```

Comment:
why moved?

Reply:
we should do flush_state_reset when async lock held

Reply:
why ? we just created this ep, no other thread could be using it..

Reply:
we do update UCP EP flags which always require blocking ASYNC

Reply:
maybe move under `ucp_worker_create_ep` INTERNAL flag handling as it was before?

Reply:
this was implemented in #6312 PR